docs/docs/dev_guide/language_api/modeling_inheritance.md

```diff
@@ -0,0 +1,150 @@
+```
+Definition(
+    uid=UUID('...'),
+    name='DataA',
+     content='...',
+     source=AaCFile(...),
+     lexemes=[...],
+     structure={
+        'schema': {
+            'inherits': ['DataMessage'],
+            'name': 'DataA',
+            'fields': [
+                {'name': 'request', 'type': 'string'}
+                {'name': 'source', 'type': 'string'}      # <-- Inherited Field
+                {'name': 'destination', 'type': 'string'} # <-- Inherited Field
+                {'name': 'content-length', 'type': 'int'} # <-- Inherited Field
+            ]
+        }
+    }
+)
+```
+
+## Looking up Inherited Definitions
+If you're interested in obtaining the definitions that are inherited by another definition, you can easily access the list of inherited definitions by acessing the `inherits` field, which returns a list of strings (definition names).
+```python
+active_context = get_active_context()
+data_a_definition = get_active_context().get_definition_by_name("DataA")
+inherited_definition_names = data_a_definition.get_inherits() # returns ['DataMessage']
+
+# This loop will iterate over all of the parent definitions, if they're in the active context.
+for definition_name in inherited_definition_names:
+    parent_definition = get_active_context().get_definition_by_name(definition_name)
+```
+
+# Modeling Inheritance in 0.1.2 or Earlier Versions
+While inheritance is not explicitly supported in version `0.1.2`, motivated users can recreate the mechanisms for their plugins.
+
+## Recreate the Schema Changes
+The first step will be alter the `schema` definition to support inheritance as mentioned in the [supporting changes](#modeling-inheritance-in-012-or-earlier-versions) section above.
+
+Create the new `DefinitionReference` definition:
+```yaml
+schema:
+  name: DefinitionReference
+  fields:
+    - name: name
+      type: string
+  validation:
+    - name: Type references exist
+      arguments:
+        - name
+```
+
+Extend the schema structure to include the new `inherits` field:
+```yaml
+ext:
+   name: SchemaInheritanceExtension
+   type: schema
+   schemaExt:
+      add:
+        - name: inherits
+          type: DefinitionReference[]
+```
+
+## Accessing the Inherited Fields
+Because `0.1.2` and earlier versions don't automatically apply inherited fields or validations, you will have to manually look them up. You'll start by getting the list of inherited definition names, look them up in the context, then extract the fields and validations from the parent definitions.
```

Comment:
```suggestion
Because `0.1.2` and earlier versions don't automatically apply inherited fields or validations, you will have to manually look them up. Start by getting the list of inherited definition names, look them up in the context, then extract the fields and validations from the parent definitions.
```